lib/adapters/stochastic.js

```diff
@@ -27,11 +29,11 @@ var Read = Juttle.proc.source.extend({
         }
         this.type = options.type;
         if (this.type && this.type !== 'metric' && this.type !== 'event') {
-            throw this.compile_error('RT-METRICS-OR-EVENTS');
+            throw new errors.compileError('RT-METRICS-OR-EVENTS');
         }
 
-        this.handleTimeOptions(options);
-        this.setDefaultTimeRange(true);
+        this.from = options.from || NOW;
+        this.to = options.to || NOW;
```

Comment:
Actually `-from / -to` are required to be supplied, but weren't consistently being set in the options passed to the adapter as opposed to in the read proc.

Inspired by this problem, I just pushed a rework of how the adapters indicate whether or not they require time options (see fe75f52) so that it's a static property as opposed to an instance property, which lets the `read` proc make sure to always pass a valid from/to as part of the adapter constructor options.